go.mod

```diff
@@ -176,8 +176,8 @@ require (
 	github.com/xeipuuv/gojsonschema v1.2.0
 	go.etcd.io/bbolt v1.3.6
 	go.etcd.io/etcd/client/v2 v2.305.1
-	go.opentelemetry.io/collector v0.43.1
-	go.opentelemetry.io/collector/model v0.43.1
+	go.opentelemetry.io/collector v0.44.0
+	go.opentelemetry.io/collector/model v0.44.0
```

Comment:
Can you also bump `github.com/open-telemetry/opentelemetry-collector-contrib/pkg/resourcetotelemetry`? There was a bugfix there that it would be nice to have on the Agent (open-telemetry/opentelemetry-collector-contrib#7523)